tests/test_modeling_common.py

```diff
@@ -2531,6 +2531,62 @@ def test_model_from_pretrained(self):
             self.assertEqual(model.config.output_hidden_states, True)
             self.assertEqual(model.config, config)
 
+    def test_model_from_pretrained_subfolder(self):
+        config = BertConfig.from_pretrained("hf-internal-testing/tiny-random-bert")
+        model = BertModel(config)
+        sum_all_params = sum(p.abs().sum() for p in model.parameters())
+
+        subfolder = "bert"
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            model.save_pretrained(os.path.join(tmp_dir, subfolder))
+
+            with self.assertRaises(OSError):
+                _ = BertModel.from_pretrained(tmp_dir)
+
+            model = BertModel.from_pretrained(tmp_dir, subfolder=subfolder)
+
+        sum_all_params_loaded = sum(p.abs().sum() for p in model.parameters())
```

Comment:
Weird to test this for me. Why not just loop over the zip of the two model `parameters()` and assert everything is equal?
(Can be put in a `check_models_are_equal` methods and reused)